### PR TITLE
XML Data Integration 

### PR DESCRIPTION
**Pull request by noadjamila**
## Description

Two scripts were created to import the data of the two xml files (institutions.xml, degreeprogramme.xml) into our database. The schema for the needed tables were added as well as the documentation of the database. 

## Jira Ticket

Link to the Jira ticket: [GI-143](https://techstudyfinder.atlassian.net/browse/GI-143)

## Author Checklist

- [x] Code follows project coding standards and conventions
- [x] Tests have been written or updated
- [x] All tests pass locally
- [x] No sensitive data or secrets are included
- [x] Documentation (README, comments) has been updated if necessary

## Reviewer Notes

Please verify the following during review:

- The implementation matches the described requirements and Jira ticket
- Code changes are logically correct and maintainable
- No unnecessary dependencies or side effects introduced
- Tests cover critical logic and edge cases
- Naming, structure, and readability meet team standards

## Additional Context (optional)

Add any background information or references that help reviewers understand the changes.

---

_Please ensure all criteria are met before merging._

### CODE REVIEW

**Review by psophis — COMMENTED** [review]
Habe eine Frage und eine Anmerkung, aber sieht gut aus! Sehr schöne Doku. Die Scripts muss ich noch reviewen, das mache ich heute noch

File: server/db/schema/degreeprogrammes.sql [hunk 1]
```diff
@@ -0,0 +1,109 @@
+-- ------------------------------
+-- Degree Programme Meta Tables
+-- ------------------------------
+
+CREATE TABLE abschlussart (
+    id TEXT PRIMARY KEY,
+    name TEXT
```
Comment (psophis) [inline server/db/schema/degreeprogrammes.sql]:
```suggestion
    name TEXT NOT NULL
```
Sollte nicht null sein dürfen, oder?

File: server/db/scripts/import_degreeprogrammes.ts [hunk 2]
```diff
@@ -0,0 +1,323 @@
...
+async function main() {
+  const client = new Client({
+    user: process.env.DB_USER,
+    host: process.env.DB_HOST,
+    database: process.env.DB_NAME,
+    password: process.env.DB_PASSWORD,
+    port: Number(process.env.DB_PORT),
+  });
+
+  await client.connect();
```
Comment (psophis) [inline server/db/scripts/import_degreeprogrammes.ts]: Den Import hiernach würde ich komplett in einem try-catch wrappen. Wenn ein Fehler auftritt, sollte man ein Rollback machen, damit keine unvollständigen Daten in der DB liegen

**Review by psophis — COMMENTED** [review]
Habe dir ein paar Kommentare geschrieben

**Review by psophis — APPROVED** [review]

**Review by psophis — APPROVED** [review]
